### PR TITLE
Add possitibility to specify an ami image for AWS

### PR DESCRIPTION
**Pull request by maximenoel8**
## What does this PR change?

Testing build images is currently done manually. 
This PR is part of an automation effort to test build imagefor AWS.
This PR adds the possibility to specify an image ami id for server and proxy in sumaform.

There are three main features :  
 - adding the possibility to declare an ami id in server and proxy host
 - adding the possibility to deploy without declaring repositories
 - adding a new ENV variable for controller to detect when we are using build image ( mandatory for running the test suite)

### CODE REVIEW

File: salt/server/testsuite.sls [hunk 1]
```diff
@@ -74,8 +72,6 @@ testsuite_salt_packages:
 {% if 'head' in grains.get('product_version') or 'uyuni-master' in grains.get('product_version') or 'nightly' in grains.get('product_version') %}
     - fromrepo: testing_overlay_devel_repo
 {% endif %}
-    - require:
-      - sls: repos
```
Comment (srbarrios) [inline salt/server/testsuite.sls:78]: About all the "require" sections removed.
I think we should not remove it. One thing is to include repos in the init.sls, the other to let the salt state knows that it has a dependency with repos.sls

File: modules/proxy/variables.tf [hunk 2]
```diff
@@ -152,4 +152,9 @@ variable "accept_all_ssl_protocols" {
...
   description = "Proxy should be prepared to be a containerized proxy"
   default     = false
-}
\ No newline at end of file
+}
+
+variable "ami_image" {
+  description = "An ami image to overwrite the default value"
+  default     = null
+}
```
Comment (srbarrios) [inline modules/proxy/variables.tf:160]: To be consisten with the definition in the null/host module, maybe we should have here:
```
variable "ami_image" {
  description = "An ami image to overwrite the default value"
  type        = string
  default     = null
}
```

File: modules/controller/variables.tf [hunk 3]
```diff
@@ -512,3 +512,8 @@ variable "no_mirror" {
   description = "Ignore mirror even if base configuration has one set"
   default     = false
 }
+
+variable "server_build_image" {
```
Comment (srbarrios) [inline modules/controller/variables.tf:516]: Personally I would choose a variable name that let us think this is a boolean.
Something like... "server_use_build_image"

File: salt/controller/bashrc [hunk 4]
```diff
@@ -78,6 +78,7 @@ export AUTH_REGISTRY_CREDENTIALS="{{ grains.get('auth_registry_username') }}|{{
 {% if grains.get('server_http_proxy') | default(false, true) %}export SERVER_HTTP_PROXY="{{ grains.get('server_http_proxy') }}" {% else %}# no server HTTP proxy defined {% endif %}
 {% if grains.get('pxeboot_image') | default(false, true) %}export PXEBOOT_IMAGE="{{ grains.get('pxeboot_image') }}" {% else %}# no PXE boot image defined {% endif %}
 export PROVIDER="{{ grains.get('provider') }}"
+export SERVER_BUILD_IMAGE="{{ grains.get('server_build_image') }}"
```
Comment (Bischoff) [inline salt/controller/bashrc:81]: Is it always defined ?

**Review by Bischoff — COMMENTED** [review]
just a question

**Review by Bischoff — CHANGES_REQUESTED** [review]
Isn't this PR missing changes to README_ADVANCED.md?

File: modules/host/main.tf [hunk 5]
```diff
@@ -20,6 +20,7 @@ module "host" {
   connect_to_base_network       = var.connect_to_base_network
   connect_to_additional_network = var.connect_to_additional_network
   image                         = var.image
+  ami_image                     = var.ami_image
```
Comment (rjmateus) [inline modules/host/main.tf:23]: I don't think we need this, as explained above

File: modules/proxy/main.tf [hunk 6]
```diff
@@ -57,6 +58,7 @@ module "proxy" {
   }
 
   image                    = var.image == "default" || var.product_version == "head" ? var.images[var.product_version] : var.image
+  ami_image                = var.ami_image
```
Comment (rjmateus) [inline modules/proxy/main.tf:61]: we don't need this variable

File: modules/server/main.tf [hunk 7]
```diff
@@ -82,6 +83,7 @@ module "server" {
 
 
   image                    = var.image == "default" || var.product_version == "head" ? var.images[var.product_version] : var.image
+  ami_image                = var.ami_image
```
Comment (rjmateus) [inline modules/server/main.tf:86]: we don't need this variable

File: modules/server/variables.tf [hunk 8]
```diff
@@ -269,3 +269,8 @@ variable "login_timeout" {
   description = "How long the webUI login session cookie is valid"
   default     = null
 }
+
+variable "ami_image" {
```
Comment (rjmateus) [inline modules/server/variables.tf:273]: we don't need this variable

**Review by rjmateus — CHANGES_REQUESTED** [review]
some changes that I think should be done

File: salt/server/testsuite.sls [hunk 9]
```diff
@@ -76,6 +81,7 @@ testsuite_salt_packages:
 {% endif %}
     - require:
       - sls: repos
+{% endif %} {# 'build_image' not in grains.get('product_version') | default('', true) #}
```
Comment (srbarrios) [inline salt/server/testsuite.sls:84]: I don't get why we need this comment.

**Pull request merged**